src/Servers/Kestrel/Core/src/KestrelServer.cs

```diff
@@ -200,13 +213,22 @@ public async Task StopAsync(CancellationToken cancellationToken)
 
             try
             {
-                var tasks = new Task[_transports.Count];
-                for (int i = 0; i < _transports.Count; i++)
+                var transportCount = _transports.Count;
+                var totalTransportCount = _transports.Count + _multiplexedTransports.Count;
```

Comment:
Nit: transportCount or _transports.Count. The usage is about half and half in this method now.

```suggestion
                var totalTransportCount = transportCount + _multiplexedTransports.Count;
```